scipy/stats/_hypotests.py

```diff
@@ -389,3 +391,214 @@ def _get_wilcoxon_distr(n):
                          "statistic is not implemented for n={}".format(n))
 
     return np.array(cnt, dtype=int)
+
+
+def _barnard_exact_compute_gammaln_combination(n):
+    """Compute all log combination of C(n, k)"""
+    gammaln_arr = gammaln(np.arange(n + 1) + 1)
+    return gammaln(n + 1) - gammaln_arr - gammaln_arr[::-1]
+
+
+BarnardExactResult = make_dataclass("BarnardExactResult", [(
+    'score_statistic', float),
```

Comment:
```suggestion
BarnardExactResult = make_dataclass("BarnardExactResult", [
    ('statistic', float),
```
These are almost always called `statistic`. It's not required, but unless you'd prefer a more specific name, I'd suggest sticking with the convention.